README.md

```diff
@@ -0,0 +1,146 @@
+  }
+]
+
+instance_group [{ kind: KIND_GPU, count: 1 }] # use KIND_CPU for CPU inference
+dynamic_batching {
+  max_queue_delay_microseconds: 5000 # can be tuned baesd on latency requirements
+}
+```
+
+Start the tritonserver with the ctranslate backend and model repository: 
+```bash
+tritonserver --backend-directory $BACKEND_INSTALL_DIR/backends --model-repository $MODEL_DIR
+```
+
+The backend is set up to use ragged, [dynamic batching](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#dynamic-batcher).
+This means you should send each input in it's own request and Triton will take care of batching, using the `dynamic_batching` and `max_batch_size` configuration to build appropriate batches, 
+for best performance on GPUs. 
+
+
+## Sending requests
+
+The backend expects token IDs as input and output, which need to be INT32 or bigger data-types. This might include beginning-of-sentence and end-of-sentence token, depending on the model. 
+No padding tokens need to be added though, the backend is taking care of padding and batching.
+
+You can use the offical Triton clients to make requests, both HTTP and gRPC protocols are supported. Here is an example for the opus-mt-en-de model in Python, using the async grpc client:
+
+```python
+import asyncio
+import tritonclient.grpc.aio
+from tritonclient.utils import np_to_triton_dtype
+from grpc import ChannelConnectivity
+from transformers import AutoTokenizer
+import logging
+import numpy as np
+import sys
+
+async def main():
+    MODEL_NAME = "opus-mt-en-de"
+    URL = "127.0.0.1:8001"
+    client = tritonclient.grpc.aio.InferenceServerClient(URL)
+    
+    en_text = sys.stdin.readline()
+    tokenizer = AutoTokenizer.from_pretrained("Helsinki-NLP/" + MODEL_NAME)
+    
+    input_ids = tokenizer(en_text, return_attention_mask=False, return_tensors="np").input_ids.astype(np.int32)
+    logging.info(f"Tokenised input: {input_ids}")
+    
+    if client._channel.get_state() == ChannelConnectivity.SHUTDOWN:
+        return
+    
+    inputs = [
+        tritonclient.grpc.aio.InferInput("INPUT_IDS", input_ids.shape, np_to_triton_dtype(input_ids.dtype)),
+    ]
+    inputs[0].set_data_from_numpy(input_ids)
+    outputs = [tritonclient.grpc.aio.InferRequestedOutput("OUTPUT_IDS")]
+
+    res = await client.infer(model_name=MODEL_NAME, inputs=inputs, outputs=outputs)
+    out_tokens = res.as_numpy("OUTPUT_IDS")
+    logging.info(f"Returned tokens: {out_tokens}")
+    translated_text = tokenizer.batch_decode(out_tokens)
+    print(translated_text)
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+    asyncio.run(main())
+
+```
+
+You can try this by saving the script to `client.py` and running `echo "How is your day going?" | python3 client.py`.
+
+## Additional model config parameters:
+
+Currently we support a subset of decoding parameters to be set in the config, and an additional custom `max_decoding_length_multiple`:
```

Comment:
Maybe a sentence to explain what `max_decoding_length_multiple` does

Reply:
done